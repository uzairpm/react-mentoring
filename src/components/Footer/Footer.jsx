import React from 'react';
import styled from 'styled-components'

const FooterSpan = styled.span`
    font-size: 1.4rem;
    font-Weight: 500;
    color: #fa6c85 !important;
`;

const Footer = (props) => (
    <footer className="container appFooter">
        <nav className="navbar navbar-dark bg-dark">
            <FooterSpan className="navbar-brand navbarItem">netflixroulette</FooterSpan>
        </nav>
    </footer>
);
export default Footer;
