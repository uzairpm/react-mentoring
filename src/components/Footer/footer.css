.navbar-dark .navbar-brand.navbarItem {
    font-size: 1.4rem;
    font-Weight: 500;
    color: #fa6c85;
}
.appFooter {
    /* position: absolute;
    bottom: 0;
    height: 30px;
    width: 100%; */
}