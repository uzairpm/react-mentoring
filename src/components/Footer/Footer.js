import React from 'react';

import './footer.css';

const Footer = (props) => (
    <footer className="container appFooter">
        <nav className="navbar navbar-dark bg-dark">
            <span className="navbar-brand navbarItem">netflixroulette</span>
        </nav>
    </footer>
);
export default Footer;
