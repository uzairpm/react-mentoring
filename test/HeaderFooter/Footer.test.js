import React from 'react';

import Footer from '../../src/components/Footer/Footer';

describe('Check Component creation', () => {
    test('Footer component', () => {
        const element = shallow(<Footer/>);
        expect(element).toMatchSnapshot();
        // console.log(element.debug());
        expect(element.text()).toEqual('netflixroulette');
        expect(element.find('span').prop('className')).toContain('navbarItem');
    });
});
