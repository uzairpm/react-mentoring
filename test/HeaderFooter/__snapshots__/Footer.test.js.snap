// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Component creation Footer component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="navbar navbar-dark bg-dark"
      >
        <span
          className="navbar-brand navbarItem"
        >
          netflixroulette
        </span>
      </nav>,
      "className": "container appFooter",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="navbar-brand navbarItem"
        >
          netflixroulette
        </span>,
        "className": "navbar navbar-dark bg-dark",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "netflixroulette",
          "className": "navbar-brand navbarItem",
        },
        "ref": null,
        "rendered": "netflixroulette",
        "type": "span",
      },
      "type": "nav",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navbar navbar-dark bg-dark"
        >
          <span
            className="navbar-brand navbarItem"
          >
            netflixroulette
          </span>
        </nav>,
        "className": "container appFooter",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="navbar-brand navbarItem"
          >
            netflixroulette
          </span>,
          "className": "navbar navbar-dark bg-dark",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "netflixroulette",
            "className": "navbar-brand navbarItem",
          },
          "ref": null,
          "rendered": "netflixroulette",
          "type": "span",
        },
        "type": "nav",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
